warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java

```diff
@@ -2278,4 +2384,219 @@ public Equality getEquality() {
+                }
+                // Attempt to fetch the default options from createDefaultOptions on an instance of the class.
+                try {
+                    QueryOptions instance = (QueryOptions) clazz.getConstructor().newInstance();
+                    DefaultOptions defaultOptions = instance.createDefaultOptions();
+                    // Update the map so that we only need to do this once for the class.
+                    defaultOptionsMap.put(className, defaultOptions);
+                    return defaultOptions;
+                } catch (NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+                    throw new RuntimeException("Failed to instantiate new instance of " + clazz + ", class does not have a default constructor", e);
+                }
+                
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException(e);
+            }
+            
+        }
+    }
+    
+    // Add the given option.
+    public static void addOption(IteratorSetting setting, String option, Object value, boolean allowBlankValues) {
+        // Determine the correct method for converting the value to a string.
+        if (value instanceof String) {
+            addOption(setting, option, value, (v) -> (String) v, allowBlankValues);
+        } else if (value instanceof Boolean) {
+            addOption(setting, option, value, (v) -> Boolean.toString((Boolean) v), allowBlankValues);
+        } else if (value instanceof Long) {
+            addOption(setting, option, value, (v) -> Long.toString((Long) v), allowBlankValues);
+        } else if (value instanceof Integer) {
+            addOption(setting, option, value, (v) -> Integer.toString((Integer) v), allowBlankValues);
+        } else if (value instanceof Double) {
+            addOption(setting, option, value, (v) -> Double.toString((Double) v), allowBlankValues);
+        } else if (value instanceof Float) {
+            addOption(setting, option, value, (v) -> Float.toString((Float) v), allowBlankValues);
+        } else if (value instanceof Short) {
+            addOption(setting, option, value, (v) -> Short.toString((Short) v), allowBlankValues);
+        } else if (value instanceof Byte) {
+            addOption(setting, option, value, (v) -> Byte.toString((Byte) v), allowBlankValues);
+        } else if (value instanceof Character) {
+            addOption(setting, option, value, (v) -> Character.toString((Character) v), allowBlankValues);
+        } else if (value instanceof Collection<?>) {
+            addOption(setting, option, value, (v) -> COMMA_JOINER.join((Iterable<?>) v), allowBlankValues);
+        } else if (value instanceof Enum<?>) {
+            addOption(setting, option, value, (v) -> ((Enum<?>) v).name(), allowBlankValues);
+        } else if (value instanceof Object){
```

Comment:
This can be simplified to the following. We do not need to check if Value is an Object if we've reached the last if statement. Additionally, we should default to using v.toString() if we did not match any other types above, and not throw an exception here. 

```suggestion
        } else {
            addOption(setting, option, value, (v) -> v.toString(), allowBlankValues);
        }